src/bp/ui-studio/src/web/components/Layout/StatusBar/index.tsx

```diff
@@ -40,191 +35,86 @@ const DEFAULT_STATE = {
   message: ''
 }
 
-class StatusBar extends React.Component<Props> {
-  pbRef: HTMLElement
-  progressBar: Line
-  state = { ...DEFAULT_STATE }
+const StatusBar: FC<Props> = props => {
+  const [state, dispatch] = React.useReducer(progressReducer, {
+    ...DEFAULT_STATE
+  })
 
-  componentDidMount() {
-    EventBus.default.on('statusbar.event', this.handleModuleEvent)
-    this.initializeProgressBar()
-    this.fetchTrainingSession()
-  }
+  useEffect(() => {
+    EventBus.default.on('statusbar.event', handleModuleEvent)
+    fetchTrainingSession()
+  }, [])
 
-  componentDidUpdate(pp, prevState) {
-    if (this.state.working) {
-      this.progressBar.animate(this.state.progress, 200)
-    } else {
-      if (this.state.progress < 1) {
-        // canceled
-        this.progressBar.set(0)
-      } else {
-        // done
-        this.resetStateTimeOut()
-      }
-    }
-  }
-
-  resetStateTimeOut = () => {
-    setTimeout(() => this.setState({ ...DEFAULT_STATE }), 2000)
-  }
-
-  shouldUpdateNLUEvent = (event): boolean => {
+  const shouldUpdateNLUEvent = (event): boolean => {
     return (
       event.type === 'nlu' &&
       event.botId === window.BOT_ID &&
-      _.get(event, 'trainSession.language') === this.props.contentLang
+      _.get(event, 'trainSession.language') === props.contentLang
     )
   }
 
-  handleModuleEvent = async event => {
-    if (this.shouldUpdateNLUEvent(event)) {
-      this.setState({ message: event.message, working: event.working, progress: event.trainSession.progress })
-    } else if (event.working && event.value && this.state.progress !== event.value) {
-      this.updateProgress(event.value) // @deprecated remove when engine 1 is totally gone
+  const handleModuleEvent = async event => {
```

Comment:
I know this isn't the goal of this pr but we should keep in mind that the status bar could eventually be used to display the progress of other task than training. It would be nice to make this generic and not nlu specific.

Reply:
Slightly changed it, it's hard to really know what to plan ahead for, but in theory, as long as we set the message, progress and set it as working, it'll show in the statusbar with progress % and message

Reply:
We can leave it as is but Let's keep this in mind